src/i18n.tsx

```diff
@@ -1,19 +1,30 @@
 import React from 'react'
 import { i18n } from '@lingui/core'
 import { I18nProvider } from '@lingui/react'
-import { detect, fromUrl } from '@lingui/detect-locale'
+import { detect, fromNavigator } from '@lingui/detect-locale'
 import { ReactNode, useEffect } from 'react'
+import { useLocale } from 'state/user/hooks'
 
-export const locales = ['en', 'pseudo-en', 'de', 'es-AR', 'es-US', 'it-IT', 'iw', 'ro', 'ru', 'vi', 'zh-CN', 'zh-TW']
+//www.codetwo.com/admins-blog/list-of-office-365-language-id/
+export enum Locales {
+  'en' = 'English',
+  'de' = 'Deutsch',
+  'es-AR' = 'español (Argentina)',
+  'es-US' = 'español (Estados Unidos)',
+  'it-IT' = 'italiano',
+  'iw' = 'Hebrew',
+  'ro' = 'română',
+  'ru' = 'русский',
+  'vi' = 'Tiếng Việt',
+  'zh-CN' = '中文 ( 中国 )',
+  'zh-TW' = '中文 ( 台灣 )',
+  'pseudo-en' = '',
+}
```

Comment:
i think they key of the locale enum should be dev readable, and the value should be the locale, e.g. `en`

a mapping from locale to label should be defined separately where it's rendered, e.g. in the switch component

e.g.
```
const LocaleLabel: {[locale in Locales]: string} = {...}
```

Reply:
Will add localelabel in a follow-up PR where i add the menu

Reply:
I'll go back to a simple array for supported locales for now. I think the 2 letter code is dev readable enough.

Reply:
In this case not about readability, rather it's about whether it's type safe (i.e. hard to accidentally break the code)